cpp/include/cuspatial/point_linestring_nearest_points.hpp

```diff
@@ -0,0 +1,174 @@
+ * to the nearest point on the (multi)linestring. See `point_linestring_nearest_points_result`
+ * for detail.
+ *
+ * The below example computes the nearest point from 2 points to 2 linestrings:
+ *
+ * ```
+ * The first pair:
+ * Point: (0.0, 0.0)
+ * Linestring: (1.0, -1.0) -> (1.0, 0.0) -> (0.0, 1.0)
+ * Nearest segment id in linestring: 1
+ * Nearest Point on LineString Coordinate: (0.5, 0.5)
+ *
+ * The second pair:
+ * Point: (1.0, 2.0)
+ * Linestring: (0.0, 0.0) -> (3.0, 1.0) -> (3.9, 4) -> (5.5, 1.2)
+ * Nearest segment id in linestring: 0
+ * Nearest Point on LineString Coordinate: (1.5, 0.5)
+ *
+ * Input:
+ * multipoint_parts_offsets: std::nullopt
+ * points_xy: {0.0, 0.0}
+ * multilinestring_parts_offsets: std::nullopt
+ * linestring_offsets:  {0, 3, 7}
+ * linestring_points_xy: {1, -1, 1, 0, 0, 1, 0, 0, 3, 1, 3.9, 4, 5.5, 1.2}
+ *
+ * Output:
+ * point_linestring_nearest_points_result{
+ *   std::nullopt,
+ *   std::nullopt,
+ *   {1, 0},
+ *   {0.5, 0.5, 1.5, 0.5}
+ * }
+ * ```
+ *
+ * The below example computes the nearest point from 3 multipoints to 3 multilinestrings:
+ *
+ * ```
+ * The first pair:
+ * MultiPoint: {(1.1, 3.0), (3.6, 2.4)}
+ * MultiLineString: {(2.1, 3.14) -> (8.4, -0.5) -> (6.0, 1.4), (-1.0, 0.0) -> (-1.7, 0.83)}
+ * Nearest Point id in MultiPoint: 1
+ * Nearest LineString id in MultiLineString: 0
+ * Nearest Segment id in LineString: 0
+ * Nearest Point on LineString Coordinate: (3.54513, 2.30503)
+ *
+ * The second pair:
+ * MultiPoint: {(10.0, 15.0)}
+ * MultiLineString: {(20.14, 13.5) -> (18.3, 14.3), (8.34, 9.1) -> (9.9, 9.4)}
+ * Nearest Point id in MultiPoint: 0
+ * Nearest LineString id in MultiLineString: 1
+ * Nearest Segment id in LineString: 0
+ * Nearest Point in LineString Coordinate: (9.9, 9.4)
+ *
+ * The third pair:
+ * MultiPoint: {(-5.0, -8.7), (-6.28, -7.0), (-10.0, -10.0)}
+ * MultiLineString: {(-20.0, 0.0) -> (-15.0, -15.0) -> (0.0, -18.0) -> (0.0, 0.0)}
+ * Nearest Point id in MultiPoint: 0
+ * Nearest LineString id in MultiLineString: 0
+ * Nearest Segment id in LineString: 2
+ * Nearest Point in LineString Coordinate: (0.0, -8.7)
+ *
+ * Input:
+ * multipoint_parts_offsets: {0, 2, 3, 6}
+ * points_xy: {1.1, 3.0, 3.6, 2.4, 10.0, 15.0, -5.0, -8.7, -6.28, -7.0, -10.0, -10.0}
+ * multilinestring_parts_offsets: {0, 2, 4, 5}
+ * linestring_offsets:  {0, 3, 5, 7, 9, 13}
+ * linestring_points_xy: {
+ *   2.1, 3.14, 8.4, -0.5, 6.0, 1.4, -1.0, 0.0,
+ *   -1.7, 0.83, 20.14, 13.5, 18.3, 14.3, 8.34, 9.1,
+ *   9.9, 9.4, -20.0, 0.0, -15.0, -15.0, 0.0, -18.0,
+ *   0.0, 0.0
+ * }
+ *
+ * Output:
+ * point_linestring_nearest_points_result{
+ *  {1, 0, 0},
+ *  {0, 1, 0},
+ *  {0, 0, 2},
+ *  {3.545131432802666, 2.30503517215846, 9.9, 9.4, 0.0, -8.7}
+ * }
+ * ```
+ *
+ * @param multipoint_geometry_offsets Beginning and ending indices for each multipoint
+ * @param points_xy Interleaved x, y-coordinate of points
```

Comment:
interleaved x and y doesn't match other cuspatial column-based APIs. We have always had the X/Y and Lon/Lat positions in separate columns, right?

Reply:
We started making this adaptation for https://github.com/rapidsai/cuspatial/issues/649. We could keep them separate but on the python side this would always require 2 scatter operations before we can call c++/

Reply:
Oh. I was under the impression that if you are using the column-based API you are using dataframes, which tend to have a separate column for each field. What does GeoPandas do here?

Reply:
A GeoSeries of a geopandas is an array of structures. There's current a discussion over the format of xy-coordinates going on at geoarrow: 
https://github.com/geopandas/geo-arrow-spec/pull/26

Reply:
We argued successfully to use xy interleaved coordinates for points to improve cache coherency and word size (coalesced memory access) availability for kernels when `GeoArrow` was designed. @isVoid has been in the process of converting our existing APIs to support `GeoArrow` more directly. I'll be following up in the above conversation today.